File: AtCoder/abc210/B.kt
private fun readLn() = readLine() ?: ""
private fun readLns() = readLn().split(" ")
private fun readInt() = readLn().toInt()

fun main() {
    val N = readInt()
    val S = readLn()
    var first = true
    for (i in S.indices) {
        if (S[i] == '1') {
            first = i % 2 == 0
            break
        }
    }
    when (first) {
        true -> print("Takahashi")
        false -> print("Aoki")
    }
}

File: AtCoder/abc210/B.swift
private func readLn() -> String {
	return readLine() ?? ""
}

private func readInt() -> Int {
	return Int(readLn()) ?? 0
}

let N = readInt()
let S = readLn()
var (first, index) = (true, 0)
for ch in S {
	if ch == Character("1") {
		first = index % 2 == 0
		break
	}
	index += 1
}
if first {
	print("Takahashi")
} else {
	print("Aoki")
}